sdks/java/extensions/sql/jdbc/src/test/java/org/apache/beam/sdk/extensions/sql/jdbc/BeamSqlLineIT.java

```diff
@@ -0,0 +1,328 @@
+                              + "       SELECT COUNT(*) as num_events,\n"
+                              + "              geo_cells.reduced_lat,\n"
+                              + "              geo_cells.reduced_lon, \n"
+                              + "              TUMBLE_START(geo_cells.event_timestamp, INTERVAL '1' SECOND)\n"
+                              + "       FROM geo_cells    \n"
+                              + "       GROUP BY geo_cells.reduced_lat,\n"
+                              + "                geo_cells.reduced_lon,\n"
+                              + "                TUMBLE(geo_cells.event_timestamp, INTERVAL '1' SECOND)\n"
+                              + "       LIMIT 2;");
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // LIMIT 2 is supposed to be parsed into 6 lines.
+    assertTrue(expectedResult.get().size() == 6);
+  }
+
+  @Test
+  public void testSlidingWindow() throws Exception {
+    ExecutorService pool = Executors.newFixedThreadPool(1);
+
+    Future<List<List<String>>> expectedResult =
+        pool.submit(
+            (Callable)
+                () -> {
+                  String[] args =
+                      buildArgs(
+                          String.format(createPubsubTableStatement, publicTopic),
+                          setProject,
+                          "SELECT COUNT(*) AS num_events,\n"
+                              + "              SUM(taxi_rides.payload.meter_increment) as revenue,\n"
+                              + "              HOP_END(\n"
+                              + "                 taxi_rides.event_timestamp, \n"
+                              + "                 INTERVAL '1' SECOND, \n"
+                              + "                 INTERVAL '2' SECOND) as minute_end\n"
+                              + "              FROM taxi_rides\n"
+                              + "              GROUP BY HOP(\n"
+                              + "                 taxi_rides.event_timestamp,\n"
+                              + "                 INTERVAL '1' SECOND, \n"
+                              + "                 INTERVAL '2' SECOND) LIMIT 2");
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  BeamSqlLine.runSqlLine(args, null, outputStream, null);
+                  return toLines(outputStream);
+                });
+
+    // LIMIT 2 is supposed to be parsed into 6 lines.
+    assertTrue(expectedResult.get().size() == 6);
+  }
+
+  /**
+   * Check if topic exists {@param attempts} times. There is 1 sec wait time between two attempts.
+   *
+   * @param attempts number of attempts. Has to be a value > 0
+   */
+  private void checkIfTopicExists(TestPubsub eventsTopic, int attempts)
```

Comment:
I will move it to TestPubsub.

I agree it seems more nature to pass int a Duration, which make it more like a timeout thing. However, there is no timeout argument for Pubsub client, so within this wait for function, I had to have while loop and sleep to achieve timeout. Therefore, number of attempts here is much more convenient (and less chance to make mistakes).

Reply:
I disagree with this argument:
 - as a test developer reading the test code I am not interested at all about how many attempts it takes, but how long the test would take;
 - as a developer writing the a test with this utility I normally also don't care about how many attempts it requires to finish, but rather how long will it run until timeout;
 - in both cases above the duration doesn't have to be exact, I usually don't care if the test times out after 00:02:01 vs 00:02:00;
 - with only number of attempts available to me I have to read the implementation to understand how long it will take;

You can still do a while loop limited by duration, e.g. something along these lines:

```
DateTime start = new DateTime();
subscriptionCount = listSubscriptions().size();
while (subscriptionCount == 0 && Seconds.secondsBetween(new DateTime(), startTime).getSeconds() < timeoutDuration.getStandardSeconds()) {
  Thread.sleep(1000);
  subscriptionCount = listSubscriptions().size();
}
```

Reply:
Gotcha. Since the Duration here does not mean exact time but rather an estimation, I can go with Duration.